src/objects.h

```diff
@@ -386,22 +393,23 @@ class Array : public HeapObject {
     *extra_bytes = 0;
   }
 
-  void fill(int from, Object* filler) {
-    int len = length();
-    for (int index = from; index < len; index++) {
-      at_put(index, filler);
-    }
-  }
+  inline void fill(int from, Object* filler);
 
  private:
   static const int LENGTH_OFFSET = HeapObject::SIZE;
   static const int HEADER_SIZE = LENGTH_OFFSET + WORD_SIZE;
 
   void _set_length(int value) { _word_at_put(LENGTH_OFFSET, value); }
-  void _initialize(int length, Object* filler) {
+
+  // Can only be called on newly allocated objects that will be either
```

Comment:
Can this be asserted easily?

Reply:
The function is also called from the compiler where there are no write barriers, so no.  I fixed the comment.